lab_2_tokenize_by_bpe/main.py

```diff
@@ -58,7 +102,24 @@ def train(
     :param num_merges: required number of new tokens
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
-
+    if not isinstance(word_frequencies, dict) or not isinstance(num_merges, int):
+        return None
+    while num_merges > 0:
+        num_of_pairs = count_tokens_pairs(word_frequencies)
+        if num_of_pairs is None:
+            return None
+        if num_merges > len(num_of_pairs):
+            num_merges = len(num_of_pairs)
```

Comment:
Why?

Reply:
I think that we can't do merges of more than the elements in the dictionary. And as I understand and we can do the merge only as long as there are free tokens that's why I did it such way

Reply:
The previously merged elements can be merged again with another element, so I need a more clear explanation

Reply:
Firstly I splited word_frequencies into pairs using count_token_pairs. Then I equated num_merges to the len(num_of_pairs) because as I understand we can do merging only in pairs (not between different pairs) that's why I think that this number can be num_merges if num_merges bigger than the number of keys (length of the dictionary). And if the free pairs are over, we must stop training. Can you help me if I'm mistaken because I'm already confused